tests/test_visitors/test_ast/test_builtins/test_strings/test_formatted_string.py

```diff
@@ -18,6 +21,16 @@
 
 f_string = "f'x + y = {2 + 2}'"
 f_empty_string = "f''"
+f_complex_f_string = """
+    f'{reverse(\"url-name\")}?{\"&\".join(\"user=\"+uid for uid in user_ids)}'
+"""
+f_variable_lookup = "f'smth {value}'"
+f_dict_lookup_str_key = "f'smth {dict_value[\"key\"]}'"
+f_list_index_lookup = "f'smth {list_value[0]}'"
+f_function_empty_args = "f'smth {user.get_full_name()}'"
+f_function_with_args = "f'smth {func(arg)}'"
+f_dict_lookup_function_empty_args = "f'smth {dict_value[func()]}'"
+f_list_slice_lookup = "f'smth {list[:]}'"
```

Comment:
Can we please also test cases like:
- `some.call().attr`
- `some.attr.call()`
- `some.first[0].attr.other`

Reply:
These should be allowed i guess?

Reply:
Nope, these all are forbidden.

Reply:
So where should the boundary be drawn here? Is all chaining prohibited, do we allow a single chained lookup, or restrict chaining to one call on `self`? E.g.
- `self.attr`
- `self.func()`
- `user.id`

Reply:
> a single chained lookup

This one resonates the best with me. What do you think?